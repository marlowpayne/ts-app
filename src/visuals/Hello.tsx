import React from 'react';
import styled from 'styled-components';

export interface Props {
  name?: string;
  enthusiasmLevel?: number;
}

const Wrapper = styled.div`
  text-align: center;
  margin: 20px;
  font-size: 48px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
`;

export const Hello = ({ name = 'World', enthusiasmLevel = 1 }: Props) => {
  const getExclamationMarks = (numChars: number) => {
    return Array(numChars + 1).join('!');
  };

  if (enthusiasmLevel <= 0) {
    throw new Error('You could be a little more enthusiastic. :D');
  }

  return (
    <Wrapper>
      <div className="greeting">
        Hello {name + getExclamationMarks(enthusiasmLevel)}
      </div>
    </Wrapper>
  );
};
