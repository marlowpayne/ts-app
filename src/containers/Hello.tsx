import React from 'react';
import { connect, Dispatch } from 'react-redux';
import { Hello as VisualComp } from '../visuals/Hello';
import { TextField } from './TextField';
import { StoreState } from '../types';
import { defaultName, updateName, Action } from '../actions';

export interface Props {
  name: string;
  onSubmit: Function;
  onEmptySubmit: Function;
}

// exporting for testing purposes
export const Visual = (props: Props) => {
  const onSubmit = (text: string) => {
    if (!text) {
      props.onEmptySubmit();
    } else {
      props.onSubmit(text);
    }
  };

  return (
    <div>
      <VisualComp name={props.name} />
      <TextField {...{ onSubmit }} />
    </div>
  );
};

const mapStateToProps = ({ person }: StoreState) => ({
  name: person.get('name')
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) =>
  ({
    onSubmit: (text: string) => dispatch(updateName(text)),
    onEmptySubmit: () => dispatch(defaultName()),
  });

export const Hello = connect(mapStateToProps, mapDispatchToProps)(Visual);
