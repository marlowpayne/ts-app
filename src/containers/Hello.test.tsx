import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { Visual, Props } from './Hello';

const testProps: Props = {
  name: 'Test',
  onSubmit: () => null,
  onEmptySubmit: () => null,
};

describe('Hello container comp', () => {
  it('shallow renders without crashing', () => {
    const el = shallow(<Visual {...testProps} />);
    expect(el.length).toEqual(1);
  });
});

describe('Hello container comp -- Snapshot', () => {
  it('captures a snapshot', () => {
    const rendered = renderer.create(<Visual {...testProps} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
