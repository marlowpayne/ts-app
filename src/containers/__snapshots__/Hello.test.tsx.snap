// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello container comp -- Snapshot captures a snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa jdlAAa"
  >
    <div
      className="greeting"
    >
      Hello 
      Test!
    </div>
  </div>
  <div>
    <input
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="submit"
    >
      Change Name
    </button>
  </div>
</div>
`;
